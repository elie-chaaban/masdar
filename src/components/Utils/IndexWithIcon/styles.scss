@import '../../../assets/styles/fonts';

.index-with-icon-container {
  color: white;

  .index-icon-container {
    padding-bottom: 5pt;
  }

  .subtitle {
    font-size: 0.5vw;
    line-height: 0.3vw;
    font-family: $secondary-font;
  }

  .text-content {
    font-size: 2.4vw;
    line-height: normal;
  }

  .prefix,
  .postfix {
    font-size: 1vw;
  }

  .text-container {
    display: flex;
    align-items: flex-end;
  }
}

.index-with-icon-container > * {
  display: flex;
  justify-content: center;
}
