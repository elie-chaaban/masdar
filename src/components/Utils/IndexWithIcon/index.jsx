import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';
import Row from 'react-bootstrap/Row';

const IndexWithIcon = ({icon, subtitle, text, prefix, postfix}) => {
  return (
    <div className="index-with-icon-container">
      <div className="index-icon-container">
        <img src={icon} alt="index-icon" style={{width: 30}} />
      </div>
      <Row className="align-items-center no-gutters">
        <span className="prefix align-self-end">{prefix}</span>
        <h3>{text}</h3>
        <span className="postfix align-self-end">{postfix}</span>
      </Row>
      <div className="subtitle">{subtitle}</div>
    </div>
  );
};

IndexWithIcon.propTypes = {
  icon: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  prefix: PropTypes.string,
  postfix: PropTypes.string
};

export default IndexWithIcon;
