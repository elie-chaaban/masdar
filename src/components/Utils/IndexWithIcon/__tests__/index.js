import React from 'react';
import {shallow} from 'enzyme';
import IndexWithIcon from '..';

describe('Index with Icon', () => {
  it('should have passed prpos', () => {
    const dom = shallow(<IndexWithIcon icon="url" subtitle="subtitle" text="text" prefix="$" postfix="%" />);
    expect(dom).toMatchSnapshot();
    expect(dom.find('img').prop('src')).toBe('url');
    expect(dom.find('.prefix').text()).toBe('$');
    expect(dom.find('.postfix').text()).toBe('%');
    expect(dom.find('h3').text()).toBe('text');
    expect(dom.find('.subtitle').text()).toBe('subtitle');
  });
});
