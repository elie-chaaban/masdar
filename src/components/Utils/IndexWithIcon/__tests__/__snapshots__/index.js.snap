// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index with Icon should have passed prpos 1`] = `
<div
  className="index-with-icon-container"
>
  <div
    className="index-icon-container"
  >
    <img
      alt="index-icon"
      src="url"
      style={
        Object {
          "width": 30,
        }
      }
    />
  </div>
  <Row
    className="align-items-center no-gutters"
    noGutters={false}
  >
    <span
      className="prefix align-self-end"
    >
      $
    </span>
    <h3>
      text
    </h3>
    <span
      className="postfix align-self-end"
    >
      %
    </span>
  </Row>
  <div
    className="subtitle"
  >
    subtitle
  </div>
</div>
`;
